mmf/trainers/lightning_trainer.py

```diff
@@ -0,0 +1,160 @@
+
+    def load(self):
+        super().load()
+        self._calculate_max_updates()
+        self._calculate_gradient_clip_val()
+        self._load_trainer()
+
+    def _load_trainer(self):
+        self.trainer = Trainer(
+            logger=False,
+            gpus=self._gpus,
+            num_nodes=self._num_nodes,
+            callbacks=self._callbacks,
+            precision=16 if self.training_config.fp16 else 32,
+            deterministic=self._deterministic,
+            benchmark=self._benchmark,
+            max_steps=self._max_updates,
+            max_epochs=self._max_epochs,
+            gradient_clip_val=self._gradient_clip_val,
+            num_sanity_val_steps=0,
+            automatic_optimization=True,
+            checkpoint_callback=False,
+            accumulate_grad_batches=self.training_config.update_frequency,
+            val_check_interval=self.training_config.evaluation_interval,
+            log_every_n_steps=self.training_config.log_interval,
+        )
+
+    def configure_device(self):
+        # TODO: @sash coming soon!
+        local_rank = self.config.device_id
+        registry.register("global_device", local_rank)
+        if self.config.distributed.init_method is not None:
+            self._distributed = True
+            self._gpus = [local_rank]
+            self.device = torch.device("cuda", local_rank)
+            # TODO: @sash set the corresponding node number
+            assert 0, "Please set the corresponding nodes"
+        elif torch.cuda.is_available():
+            self._gpus = [0]
+            self._num_nodes = 1
+        else:
+            self._gpus = None
+            self._num_nodes = 1
+
+    def configure_seed(self):
+        default_deterministic = False
+        seed = self.config.training.seed
+
+        self._deterministic = default_deterministic
+        if seed is not None:
+            self._deterministic = True
+
+        self._benchmark = False
+
+    def load_datasets(self):
+        logger.info("Loading datasets")
+        data_module = LightningDataModule(self.config)
+        data_module.prepare_data()
+        self.data_module = data_module
+
+    def load_model(self):
+        logger.info("Loading models")
+
+        attributes = self.config.model_config[self.config.model]
+        if isinstance(attributes, str):
+            attributes = self.config.model_config[attributes]
+        with omegaconf.open_dict(attributes):
+            attributes.model = self.config.model
+
+        self.model = build_model(attributes)
+        self.model.is_pl_enabled = True
+
+    def load_optimizer(self):
+        logger.info("Loading optimizer")
+        if self.trainer:
+            pass
+            # TODO: sash, this is a problem, needs fix
+            # at this point the model's configure_optimizer
+            # function isnt called yet, so this will break.
+            # self.optimizer = self.trainer.optimizers[0]
+            # we need an assigned optimizer for checkpoint
```

Comment:
You can also refactor the code to expect the trainer to have optimzer property.

Reply:
Addressed in the next PR.